src/dbnode/storage/index/block.go

```diff
@@ -498,6 +486,13 @@ func (b *block) queryWithSpan(
 		docsPool.Put(batch)
 	}()
 
+	select {
+	case <-ctx.MustGoContext().Done():
+		// indexNs will log something useful.
+		return false, ErrCancelledQuery
+	default:
+	}
```

Comment:
If we already check for cancellation at the query worker pool level before the start of this query starts executing, do we need one as well before the batches do you think?

Probably fine to be here, but also probably would be fine to omit this too since we check it before we get the token and so we probably momentarily check this prior to now (haven't started searches yet I believe?).

Reply:
yea I added this to get a concurrency test to actually return a timeout, since the test had less blocks than the batch :/  not a good reason. I'll remove and get the test working another way. 

it might not be a bad idea to check inside the iterator when the len(batch) == 0, since that would mean search has run